vocab.py

```diff
@@ -20,9 +28,102 @@ class ValidDataSourceName(Enum):
     Each data source name should be defined in ValidDataSourceName.
     NAME_OF_DATA_SOURCE = auto()
     """
-    
+
     CANVAS_API = auto()
     CANVAS_ZOOM_MEETINGS = auto()
     KALTURA_API = auto()
     UNIZIN_DATA_PLATFORM_EVENTS = auto()
     UNIZIN_DATA_WAREHOUSE = auto()
+
+    def __str__(self) -> str:
+        '''
+        Return value name without `ValidDataSourceName` class name.
+
+        :return: Name of enum value.
+        '''
+        return str(self.name)
+
+
+class DataSourceStatus:
+    '''
+    Each data source process must return an instance of this class, which tells the name of the
+    data source and the time at which data from it was last updated.
+    '''
+
+    def __init__(
+            self,
+            data_source_name: ValidDataSourceName = NotImplemented,
```

Comment:
I recently picked up this usage of `NotImplemented` from another module.  I think it was the Kaltura API.  

It appears to be used there because `None` could be a valid value of the arguments.  `NotImplemented` is used to indicate a value wasn't given for an argument.

However, the more I think about this, it's obvious that `None` is not a valid value for this code, so there's no reason not to use that.  Also, probably `False` would be a better indicator than `NotImplemented` in a lot of other cases.

For the sake of harmony, I'll change the default values here to `None`.  Then I could set the type hint to `Optional[ValidDataSourceName]`, although I really, really dislike the name `Optional` for this purpose, but I also dislike the wordiness of a `Union` that includes `None`.

In these cases, since `None` is just a placeholder to indicate a value wasn't given, I feel like the best hint is the **_desired_** type.